src/poll-until-promise.ts

```diff
@@ -171,6 +186,10 @@ export class PollUntil {
         this._log(`then done waiting with result: ${result}`);
       })
       .catch((err: Error) => {
+        if (err instanceof AbortError) {
+          this._log(`aborted with err: ${err.cause}`);
+          return this.reject?.(err.cause);
```

Comment:
Shouldn't we pass the whole error?
```suggestion
          return this.reject?.(err);
```

Reply:
It's possible I'm missing something, but I thought we'd want to unwrap it so we could preserve the stack trace of the underlying error...?

Reply:
i think i missed something. looking at the AbortError tests I see the full error. so we're good.